src/components/LoginForm.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { FC, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { fetchTokens } from '../redux/auth/actions';
+import { authSelector } from '../redux/auth/selectors';
+import { LoginData } from '../types/auth';
+import Error from './Error';
+import Loader from './Loader';
+
+const LoginForm: FC = () => {
+  const { tokens, error, loading } = useSelector(authSelector);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [name, setName] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+
+  function handleSubmit(event: React.MouseEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const loginData: LoginData = {};
+    loginData.username = name;
+    loginData.password = password;
```

Comment:
it is more common to do this inline
const loginData: LoginData = {
username: name,
password
};